store/tikv/backoff.go

```diff
@@ -136,7 +136,7 @@ func (t backoffType) String() string {
 	return ""
 }
 
-func (t backoffType) TError() *terror.Error {
+func (t backoffType) TError() error {
```

Comment:
`*terror.Error` ?

Reply:
woh, `terror` is one of two logic change in 3rd commit

in previous `terror.Error` maintain stack frame by itself via `file`, `line` https://github.com/pingcap/tidb/pull/7151/commits/07a6f2968a11114fe7cb4865a783b238f5e54b93#diff-33cf41394e16bb72bf075a6278c80d52L225, so `GenXXX` can return `*terror.Error` directly.

but after this patch, `GenXXX` will return a wrap of `terror.Error`(pkg's WithStack),  so we need return a error interface, and in test, we use `errors.Cause` to check.